src/main/java/org/spongepowered/api/service/permission/Subject.java

```diff
@@ -183,25 +216,28 @@ default boolean hasPermission(final String permission) {
      * <p>This method is likely to be called frequently, so it is desirable
      * that implementations cache the results to method calls.</p>
      *
-     * @param contexts The contexts to check for permissions in
      * @param permission The permission to check
+     * @param cause The cause to gather context from.
      * @return The tristate result of the check
      */
-    Tristate getPermissionValue(Set<Context> contexts, String permission);
+    Tristate getPermissionValue(String permission, Cause cause);
```

Comment:
I think it would be useful to retain methods which allow lookup by `Set<Context>` so users can bypass ContextCalculators if they want to.